File: vokimi_api/Src/enums/Language.cs
﻿namespace vokimi_api.Src.enums
{
    public enum Language
    {
        Eng,
        Rus,
        Spa,
        Ger,
        Fra,
        Other
    }
    public static class LanguageExtensions
    {
        public static string GetId(this Language language) => language switch {
            Language.Eng => "eng",
            Language.Rus => "rus",
            Language.Spa => "spa",
            Language.Ger => "ger",
            Language.Fra => "fra",
            Language.Other => "other",
            _ => throw new NotImplementedException()
        };
        public static Language? FromId(string? id) => id switch {
            "eng" => Language.Eng,
            "rus" => Language.Rus,
            "spa" => Language.Spa,
            "ger" => Language.Ger,
            "fra" => Language.Fra,
            "other" => Language.Other,
            _ => null
        };

    }
}

File: vokimi_frontend/src/ts/enums/Language.ts
export enum Language {
    Eng = "eng",
    Rus = "rus",
    Spa = "spa",
    Ger = "ger",
    Fra = "fra",
    Other = "other"
}

export namespace LanguageUtils {
    export function fromId(id: string): Language {
        for (const key in Language) {
            if (Language[key as keyof typeof Language] === id) {
                return Language[key as keyof typeof Language];
            }
        }
        return Language.Other;
    }
    export function getFullName(language: Language): string {
        switch (language) {
            case Language.Eng:
                return "English";
            case Language.Rus:
                return "Русский";
            case Language.Spa:
                return "Español";
            case Language.Ger:
                return "Deutsch";
            case Language.Fra:
                return "Français";
            case Language.Other:
                return "Other";
            default:
                return "Unknown Language";
        }
    }
}